src/agent/onefuzz-supervisor/src/commands.rs

```diff
@@ -70,12 +70,37 @@ pub async fn add_ssh_key(key_info: SshKeyInfo) -> Result<()> {
                 .context("icalcs failed to run")?;
             if !result.status.success() {
                 bail!(
-                    "removing 'NT AUTHORITY/Authenticated Users' permissions to '{}' failed: {:?}",
+                    "checking permissions failed: '{}' failed: {:?}",
                     admin_auth_keys_path.display(),
                     result
                 );
             }
 
+            let stdout = String::from_utf8_lossy(&result.stdout).to_string();
+
+            if stdout.contains(&admins) {
+                let result = Command::new("icacls.exe")
+                    .arg(&admin_auth_keys_path)
+                    .arg("/remove")
+                    .arg(&admins)
+                    .stdin(Stdio::null())
+                    .stdout(Stdio::piped())
+                    .stderr(Stdio::piped())
+                    .spawn()
+                    .context("icacls remove failed to start")?
+                    .wait_with_output()
+                    .await
+                    .context("icalcs remove failed to run")?;
+                if !result.status.success() {
```

Comment:
would it make sense to retry the operation? I am thinking we could be hitting some transient issue

Reply:
I didn't add this due to transient issue, but this prevents the failure that was induced by a different Azure VM extension that added an SSH key.